src/inputs/AddressInput/index.tsx

```diff
@@ -24,95 +31,137 @@ type AddressInputProps = {
   disabled?: boolean;
   onChangeAddress: (address: string) => void;
   getAddressFromDomain?: (name: string) => Promise<string>;
+  customENSThrottleDelay?: number;
   showLoadingSpinner?: boolean;
 } & TextFieldInputProps;
 
 function AddressInput({
   name,
   address,
   networkPrefix,
-  showNetworkPrefix,
+  showNetworkPrefix = true,
   disabled,
   onChangeAddress,
   getAddressFromDomain,
+  customENSThrottleDelay,
   showLoadingSpinner,
   InputProps,
+  inputProps,
+  hiddenLabel = true,
   ...rest
 }: AddressInputProps): ReactElement {
-  const [prefix, setPrefix] = useState(networkPrefix);
   const [isLoadingENSResolution, setIsLoadingENSResolution] = useState(false);
-
+  const defaulInputValue = addPrefix(address, networkPrefix, showNetworkPrefix);
+  const inputRef = useRef({ value: defaulInputValue });
   const throttle = useThrottle();
 
-  // ENS resolution
+  // we checksum & include the network prefix in the input if showNetworkPrefix is set to true
+  const updateInputValue = useCallback(
+    (value = '') => {
+      if (inputRef.current) {
+        const checksumAddress = checksumValidAddress(value);
+        inputRef.current.value = addPrefix(
+          checksumAddress,
+          networkPrefix,
+          showNetworkPrefix
+        );
+      }
+    },
+    [networkPrefix, showNetworkPrefix]
+  );
+
+  // ENS name resolution
   useEffect(() => {
-    const resolveDomainName = async (ENSName: string) => {
-      try {
-        setIsLoadingENSResolution(true);
-        const address = (await getAddressFromDomain?.(ENSName)) as string;
-        onChangeAddress(checksumValidAddress(address));
-        setPrefix(showNetworkPrefix ? networkPrefix : '');
-      } catch (e) {
-        onChangeAddress(checksumValidAddress(ENSName));
-        setPrefix('');
-      } finally {
-        setIsLoadingENSResolution(false);
+    const resolveDomainName = async () => {
```

Comment:
perhaps make sense an useCallback outside the effect?

Reply:
done!